src/app/services/actions/autocomplete-action-creators.ts

```diff
@@ -0,0 +1,91 @@
+import { IAction } from '../../../types/action';
+import { IRequestOptions } from '../../../types/request';
+import { AUTOCOMPLETE_FETCH_ERROR, AUTOCOMPLETE_FETCH_PENDING, AUTOCOMPLETE_FETCH_SUCCESS } from '../redux-constants';
+
+export function fetchAutocompleteSuccess(response: object): IAction {
+  return {
+    type: AUTOCOMPLETE_FETCH_SUCCESS,
+    response,
+  };
+}
+
+export function fetchAutocompleteError(response: object): IAction {
+  return {
+    type: AUTOCOMPLETE_FETCH_ERROR,
+    response,
+  };
+}
+
+export function fetchAutocompletePending(): any {
+  return {
+    type: AUTOCOMPLETE_FETCH_PENDING
+  };
+}
+
+export function fetchAutoCompleteOptions(url: string): Function {
+  return async (dispatch: Function, getState: Function) => {
+    const devxApi = getState().devxApi;
+    const sampleQuery = getState().sampleQuery;
+    const headers = {
+      'Content-Type': 'application/json',
+    };
+
+    let permissionsUrl = `${devxApi}/openapi`;
+    permissionsUrl = `${permissionsUrl}?url=/${url}&style=geautocomplete&format=json`;
+
+    const options: IRequestOptions = { headers };
+
+    dispatch(fetchAutocompletePending());
+
+    try {
+      const response = await fetch(permissionsUrl, options);
+      if (response.ok) {
+        const autoCompleteOptions = await response.json();
+        const params = {
+          options: autoCompleteOptions,
+          url,
+          verb: sampleQuery.selectedVerb.toLowerCase()
+        };
+        const reduced = getReducedVersion(params);
```

Comment:
The variable name `reduced` and the function name `getReducedVersion` do not communicate their intent. Let's use more descriptive names.

Reply:
good catch

Reply:
moving this to it's own file and renaming it parseOpenApiResponse